### PR TITLE
control-service: Add python_version to Execution API

### PR DESCRIPTION
**Pull request by mivanov1988**
# Why
As part of the initiative to add support for multiple Python versions,  we need to extend the Execution API to return python_version for each individual execution.

# What
Added job_python_version column to the data_job_execution table and job_python_version field to all DTOs related to the Execution API.

# Test done
Unit tests. IT test will be added later since the feature for adding support for multiple Python versions is not ready for e2e testing.

Signed-off-by: Miroslav Ivanov miroslavi@vmware.com

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/DataJobExecution.java [hunk 1]
```diff
@@ -50,6 +50,8 @@ public class DataJobExecution {
 
   private String jobVersion;
 
+  private String jobPythonVersion;
+
```
Comment (ivakoleva) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/DataJobExecution.java:54]: Should the be api.yaml changes?
VEP-1739 contains only Deployment API change, may include Execution API.

**Review by ivakoleva — APPROVED** [review]
LGTM

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/ToApiModelConverter.java [hunk 2]
```diff
@@ -202,6 +202,7 @@ public static DataJobExecution jobExecutionToConvert(
             new DataJobDeployment()
                 .vdkVersion(jobExecutionToConvert.getVdkVersion())
                 .jobVersion(jobExecutionToConvert.getJobVersion())
+                .pythonVersion(jobExecutionToConvert.getJobPythonVersion())
```
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/ToApiModelConverter.java:205]: It has already changed by @

> LGTM

Thanks for the review, @ivakoleva!

**Comment by mivanov1988** [comment]
> LGTM

Thanks for the review, @ivakoleva!

File: projects/control-service/projects/pipelines_control_service/src/main/resources/db/migration/V20230410140000__add_column_job_python_version_to_data_job_execution_table.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+alter table if exists data_job_execution
+    add column if not exists job_python_version varchar;
```
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/resources/db/migration/V20230410140000__add_column_job_python_version_to_data_job_execution_table.sql:2]: In the beginning, I thought about that. However, it is not an indexed column and also it is it in full control of the Control Service. If you think it makes sense, I'll add a limit.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/ToApiModelConverter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/ToApiModelConverter.java:205]: It has already been changed by @doks5. 

https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/model/apidefs/datajob-api/api.yaml#L1088

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/DataJobExecution.java (hunk 1 above)
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/model/DataJobExecution.java:54]: It makes sense to me. I'll add a sentence for that in a separate PR.

**Review by murphp15 — APPROVED** [review]

**Pull request merged**